### PR TITLE
implementation_of_registration_and_login_apis

### PR DESCRIPTION
**Pull request by AnkushM123**
I have implemented registration and login apis for user.

### CODE REVIEW

**Review by ix-pankajk — CHANGES_REQUESTED** [review]
Add git ignore file

File: apis/login.js [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  if (req.body.age < 0 || req.body.age > 60) {
+    res.status(400).send("age must be within 60");
+    return
+
+  }
+
+  if (!validator.isEmail(req.body.email)) {
+    res.status(400).send("please enter valid email");
+    return
+  }
```
Comment (ix-pankajk) [inline apis/login.js]: remove unwanted return statement

File: apis/login.js [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+  const user = {"id":data[0]._id}
+  
+  if (isMatch) {
+    const token = jwt.sign(user, secretKey, { expiresIn: '1h' });
+    res.json({ token });
+  }
+}
+catch(err){
+console.log(err);
+}
```
Comment (ix-pankajk) [inline apis/login.js]: remove console logs

File: core/services/user-service.js [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
-module.exports = { getUser};
\ No newline at end of file
+const createUser = async function (user) {
+    try {
+        const data = await userModel.create(user)
+        return data;
+    } catch (err) {
+        console.log(err);
+    }
+}
```
Comment (ix-pankajk) [inline core/services/user-service.js]: remove console log from all files

**Review by ix-pankajk — CHANGES_REQUESTED** [review]

File: apis/login.js [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+    updatedBy: req.body.updatedBy,
+    avatar: req.file.path
+  })
+  employee.roles
+  let data = await userService.createUser(employee);
+  res.send(data);
+}
+
+const secretKey = 'dcsgjvjcddsdhvscskhadafsrgvrsgrf';
+
```
Comment (ix-pankajk) [inline apis/login.js]: add constant file and do not use hardcoded values

**Review by ix-pankajk — CHANGES_REQUESTED** [review]

File: apis/login.js (hunk 1 above)
Comment (AnkushM123) [reply apis/login.js]: done,removed all unwanted return statement

File: apis/login.js (hunk 2 above)
Comment (AnkushM123) [reply apis/login.js]: done,removed console logs

File: apis/login.js (hunk 4 above)
Comment (AnkushM123) [reply apis/login.js]: done, added every hardcoded values in constant file

File: apis/auth.js [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+  jwt.verify(token, secretKey, (err, user) => {
+
+    if (err) {
+      return res.status(403).json({ error: 'Forbidden: Invalid token' });
+    }
+    req.user = user;
+    next();
+  });
+};
```
Comment (ix-pankajk) [inline apis/auth.js]: Move to the middlewares

File: apis/image_api.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+const multer = require('multer');
+const path = require('path');
```
Comment (ix-pankajk) [inline apis/image_api.js]: instead of _ use - in file name.
file name should be file-upload.js

File: core/constant/passwordRegex.js [hunk 7]
```diff
@@ -0,0 +1,6 @@
+const validatePassword = () => {
+    let regex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$/;
```
Comment (ix-pankajk) [inline core/constant/passwordRegex.js]: regex should be const

File: dtos/registerValidation.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+const yup = require('yup');
+const passwordRegex=require('../core/constant/passwordRegex');
+
+const registerSchema = yup.object().shape({
+    name: yup.string().required('Name is required'),
+    address: yup.object({
+        flat_details: yup.string().required('Flat deails are required').max(100, 'Flat details must be less than or equal to 100 characters'),
+        area: yup.string().required('area deails are required').max(100, 'Area details must be less than or equal to 100 characters'),
+        landmark: yup.string().required('landmark deails are required').max(100, 'Landamark details must be less than or equal to 100 characters'),
```
Comment (ix-pankajk) [inline dtos/registerValidation.js]: Include addressLineOne, addressLinetwo, city, state, country, and postal code in the address object

**Review by ix-pankajk — CHANGES_REQUESTED** [review]

**Review by ix-pankajk — APPROVED** [review]

**Pull request merged**